### PR TITLE
tutorials: add a tutorial for loggert and message

### PR DESCRIPTION
**Pull request by xieyuschen**
Add a tutorial as proposed https://github.com/co-log/co-log/issues/84 for message and loggert.
This PR includes:
- [x] Same example but with LoggerT monad and rich message
- [x] Basic example with LoggerT but not printing `threadId`

Regards

### CODE REVIEW

File: tutorials/3-loggert-with-message/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Message and LoggerT
+
+The previous tutorial shows how to use `loggert` with simple message.
```
Comment (vrom911) [inline tutorials/3-loggert-with-message/README.md:3]:
```suggestion
The previous tutorial shows how to use `LoggerT` with simple message.
```

File: tutorials/3-loggert-with-message/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+The message is parametrized by the Severity type through `Msg`, which is a general logging message data type.
+
+```idris
+type Message = Msg Severity
+data Msg sev = Msg
+    { msgSeverity :: !sev
+    , msgStack    :: !CallStack
+    , msgText     ::  Text
+    }
+````
```
Comment (vrom911) [inline tutorials/3-loggert-with-message/README.md:78]:
```suggestion
```
```

File: tutorials/3-loggert-with-message/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+Hence, besides printing the callstack as the `loggert`` with simple message tutorial shows, messages support print severity and thread id as well.
```
Comment (vrom911) [inline tutorials/3-loggert-with-message/README.md:80]:
```suggestion
Hence, besides printing the callstack, as the `LoggerT` with simple message tutorial shows, this example supports printing severity and thread id as well.
```

File: tutorials/3-loggert-with-message/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+example2 = do
+    logError "Mayday! Mayday! Here is an example of error log\n"
+```
+
+The `WithLog` constraint has three type parameters: the application environment,
+the type of the message and the monad. The actions constraint by `WithLog` could be used as `LoggerT`.
+
+## Message
+
+The message is parametrized by the Severity type through `Msg`, which is a general logging message data type.
```
Comment (vrom911) [inline tutorials/3-loggert-with-message/README.md:69]:
```suggestion
The message is parametrized by the `Severity` type through `Msg`, which is a general logging message data type.
```

File: tutorials/3-loggert-with-message/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+example1 = do
+    logInfo "this is a demo log for message!"
+
+example2 :: WithLog env Message m => m ()
+example2 = do
+    logError "Mayday! Mayday! Here is an example of error log\n"
+```
+
+The `WithLog` constraint has three type parameters: the application environment,
+the type of the message and the monad. The actions constraint by `WithLog` could be used as `LoggerT`.
```
Comment (vrom911) [inline tutorials/3-loggert-with-message/README.md:65]:
```suggestion
The `WithLog` constraint has three type parameters:
 * `env` – application environment
 * `msg` – type of the message
 * `m` – monad 
 
 The actions constraint by `WithLog` could be used as `LoggerT`.
```

**Review by vrom911 — COMMENTED** [review]
Thanks! Added few small comments 🙂

**Comment by xieyuschen** [comment]
> Thanks! Added few small comments 🙂 

hi @vrom911 thank you for reviewing! I applied your suggestions and squash them. If you have any concerns about the PR, kindly let me know:)

**Review by alaendle — APPROVED** [review]
Many thanks for your effort; to be honest I haven't looked in details yet - but what I've seen on a quick way looks good 👍

**Comment by alaendle** [comment]
@xieyuschen Many thanks; do you have interest to be added as a collaborator - this would give you at least some more access rights ;-) If not that's also fine; we are happy about any contribution from your side.

**Comment by xieyuschen** [comment]
> @xieyuschen Many thanks; do you have interest to be added as a collaborator - this would give you at least some more access rights ;-) If not that's also fine; we are happy about any contribution from your side.

Sure, it's an honor to be a collaborator:) 

Regards

**Pull request merged**